community/en/neural_network_from_scratch_tensorflow_core.ipynb

```diff
@@ -0,0 +1,579 @@
+    "    Z = self.forward_prop(x_batch)\n",
+    "    cost = tf.reduce_mean(\n",
+    "        tf.nn.softmax_cross_entropy_with_logits_v2(logits=Z, labels=y_batch))\n",
+    "    return cost"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "KVjT6C2SqmH0"
+   },
+   "source": [
+    "## Evaluating the Model\n",
+    "As we train the model, we may want to look at its performance on the training data. The cost computed in the previous step can be a good indicator. Another indicator could be the accuracy on the training set. We will write a function to calculate this accuracy and it will be used again after the training is complete to calculate accuracy of the model on the test set. We can use `tf.argmax` to find the classes with the highest probability score and compare it with the given labels."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "KNSJqjY7qnCe"
+   },
+   "outputs": [],
+   "source": [
+    "class NNModel(NNModel):\n",
+    "  def evaluate(self, x_eval, y_eval, x, y):\n",
+    "    correct = tf.equal(tf.argmax(self.forward_prop(x), 1), tf.argmax(y, 1))\n",
+    "    acc = tf.reduce_mean(tf.cast(correct, 'float'))\n",
+    "    accuracy = acc.eval({x: x_eval, y: y_eval})\n",
+    "    return accuracy"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "8c4bNlGLrd5T"
+   },
+   "source": [
+    "## The Training Loop\n",
+    "Finally, we will write a function to run the training loop. This function will take arguments for the training set, the test set, number of epochs and batch size. We will first initialize the weights and biases, then create placeholders for the examples and labels to be used for forward propagation, computing costs and for evaluating the accuracies. We will use the _Adam Optimizer_ algorithm provided in TensorFlow.\n",
+    "\n",
+    "The training loop itself needs to run within a TensorFlow session. In this loop, we will calculate costs for all the batches by running the optimizer to minimize the costs. After each epoch, cost for that epoch will be printed out along with the evaluated accuracy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "K8Sh1UaQrc5D"
+   },
+   "outputs": [],
+   "source": [
+    "class NNModel(NNModel):\n",
+    "  def train(self, x_train, y_train, x_test, y_test, epochs, batch_size):\n",
+    "    self.initialize_params()\n",
+    "    m = x_train.shape[0]\n",
+    "\n",
+    "    x = tf.placeholder('float', [None, self.num_features])\n",
+    "    y = tf.placeholder('float', [None, self.num_classes])\n",
+    "\n",
+    "    cost = self.compute_cost(x, y)\n",
+    "    optimizer = tf.train.AdamOptimizer().minimize(cost)\n",
+    "\n",
+    "    with tf.Session() as sess:\n",
```

Comment:
Pretty neat notebook!
Can you update it to use TF 2.0? We do want to include these APIs in the examples.
See: https://www.tensorflow.org/alpha

Reply:
I can update it, yes! I have a question though: Since TF 2.0 uses Eager Execution by default, should I update the code to utilize Eager Execution or should I use sessions as it's currently written and disable Eager Execution?

My guess would be to use Eager Execution since that seems to be the recommended way going forward. Am I right?

Reply:
There is not :)
Eager by default with [tf.function](https://www.tensorflow.org/alpha/guide/autograph) (if you need it).
I would set up the notebooks like the tutorials in the /alpha section (pip install tf2 if you need to). That is definitely the way forward.
Thank you!

Reply:
I updated it to TF 2.0